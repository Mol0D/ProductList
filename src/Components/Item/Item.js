import React from 'react';
import './Item.css';
import PropTypes from 'prop-types';
import {Card} from 'react-bootstrap';
const Item = props =>{
    
        const {name, img, price, stars, link, brand} = props;
        return(
            
            <Card className="item">
                <Card.Img variant="top" src={img} width="320px" height="240px"/>
                <Card.Body>
                    <Card.Title className="text">{name}</Card.Title>
                        <Card.Text>
                        Price: {price}
                        </Card.Text>
                        <Card.Text>
                        Stars: {stars}
                        </Card.Text>
                        <Card.Text>
                        Brand: {brand}
                        </Card.Text>
                        <Card.Link href={link}>Go to shop</Card.Link>
                </Card.Body>
            </Card>
        )
    }

Item.propTypes = {
    name: PropTypes.string,
    img: PropTypes.string,
    price: PropTypes.number,
    stars: PropTypes.number,
    link: PropTypes.string,
    brand: PropTypes.string
}

export default Item;