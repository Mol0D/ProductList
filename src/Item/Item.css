.item{
    width: 340px;
    border: 1px solid gold;
    margin: 7px 0;
    padding: 10px;

}

.name{
    text-align: center;
    font-size: 20px;
    padding: 0 10px;

}

.img{
    width: 320px;
    height: 240px;
    display: block;
    margin: 0 auto;
}

.link{
    display: block;
    width: 150px;
    padding: 10px 0;
    background: red;
    text-align: center;
    text-decoration: none;
    color: white;
    margin: 5px auto 0;
}

.infoName{
    text-align: center;
    padding-top: 7px;
    font-size: 20px;
    font-style: italic;
}

.info{
    padding: 5px 0;
}