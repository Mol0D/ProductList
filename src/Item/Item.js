import React from 'react';
import './Item.css';


class Item extends React.Component{
    constructor(props){
        super(props);
        this.state={

        }
    }


    render(){
        const {name, img, price, stars, link, brand} = this.props;
        return(
            <div className='item'>
                <p className='name'>{name}</p>
                <img src={img} className='img'/>
                <a className="link" href={link}>Go to shope!</a>
                <p className='infoName'>Inforamtion</p>
                <p className="info">Price: {price}</p>
                <p className="info">Stars: {stars}</p>
                <p className="info">Brand: {brand}</p>
            </div>
        )
    }
}
export default Item;