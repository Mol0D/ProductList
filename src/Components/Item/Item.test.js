import React from 'react';
import renderer from 'react-test-renderer';
import Item from './Item';
import {shallow, mount} from 'enzyme';

it("without component", () => {
    const wrapper = shallow(<Item/>);
    expect(wrapper).toMatchSnapshot();
})

it("with component", () =>{
    const props ={
        name: 'AAA',
        img: '/src',
        price: 15.99,
        stars: 4.5,
        link: 'www',
        brand: 'GOD'
    }, 
    wrapper = shallow(<Item {...props}/>);
    expect (wrapper).toMatchSnapshot()
})

it("correct Prop Name", ()=>{
    const wrapper = mount(<Item name="AAA"/>);
    expect(wrapper.prop('name')).toEqual('AAA')
}) 

it("correct Prop Brand", () =>{
    const wrapper = mount(<Item brand="GOD"/>);
    expect(wrapper.prop('brand')).toEqual('GOD')
})

it("correct Prop Price", () =>{
    const wrapper = mount(<Item price={15.99}/>);
    expect(wrapper.prop('price')).toEqual(15.99)
})

it("correct Prop Stars", () =>{
    const wrapper = mount(<Item stars={4.5}/>);
    expect(wrapper.prop('stars')).toEqual(4.5)
})

it("correct Prop Img", () =>{
    const wrapper = mount(<Item img='/src'/>);
    expect(wrapper.prop('img')).toEqual('/src')
})

it("correct Prop Link", () =>{
    const wrapper = mount(<Item link='www'/>);
    expect(wrapper.prop('link')).toEqual('www')
})
