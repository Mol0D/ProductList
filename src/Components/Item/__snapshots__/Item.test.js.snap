// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`with component 1`] = `ShallowWrapper {}`;

exports[`without component 1`] = `ShallowWrapper {}`;
